python/ray/tune/trial_runner.py

```diff
@@ -570,7 +570,7 @@ def _start_trial(trial: Trial) -> bool:
             if self.is_finished():
                 self._server.shutdown()
         with warn_if_slow("on_step_end"):
-            self.trial_executor.on_step_end(self)
+            self.trial_executor.on_step_end(self.get_trials())
```

Comment:
so concretely, given that we know other people are extending/relying on the trial executor interface, we should raise a better warning here right (if this fails)?

Reply:
Good point, made another try at the problem. 

The current approach I took is try catch the AttributeError exception in TrialRunner (caller side) while urging user to migrate off of using TrialRunner in the API. However, as noted in doc string and user facing message, this is not bullet proof. Could very likely lead to weird behaviors depending on how user implements Executor interface. 

I also removed the warning message in RayTrialExecutor about the API change, under the assumption that TrialRunner is going to be the only caller to RayTrialExecutor. 

Overall, I find it much easier to extend an API than restricting it (like in this case, restricting its scope from TrialRunner to just the trials). Let me know what you folks think!